File: LuaCode/Util/MemUtil.lua
--=========================================================================
-- Copyright (C) , 金七情(Loong) jinqiqing@qq.com
-- Created by Loong on 2018-06-19 17:56:26
-- 内存工具
--=========================================================================

MemUtil = {Name = "MemUtil"}

local My = MemUtil

--上一次可用内存
My.lastAvaiMem = 0

--内存递增/减少快照
--des(string):描述
function My.Snap(des)
	local avaiMem = Device.AvaiMem
	local lastAvaiMem = My.lastAvaiMem
	des = des or "app"
	if lastAvaiMem < 1 then
		iTrace.Log("Loong", des, " AvaiMem:", avaiMem, "M")
	else
		local dif, str = 0, nil
		if avaiMem < lastAvaiMem then
			dif = lastAvaiMem - avaiMem
			str = " ,decreased:"
		else
			dif = avaiMem - lastAvaiMem
			str = " ,increased:"
		end
		iTrace.Log("Loong", des, " AvaiMem:", avaiMem, "M", str, dif)
	end
	My.lastAvaiMem = avaiMem
end

return My

File: Pro/Assets/Script/Client/Util/MemUtil.cs
/*=============================================================================
 * Copyright (C) 2018, 金七情(Loong) jinqiqing@qq.com
 * Created by Loong on 2018/6/19 19:29:52
 ============================================================================*/

using System;
using System.Collections;
using System.Collections.Generic;

namespace Loong.Game
{
    /// <summary>
    /// 内存工具
    /// </summary>
    public static class MemUtil
    {
        #region 字段
        /// <summary>
        /// 上一次可用内存
        /// </summary>
        private static int lastAvaiMem = 0;
        #endregion

        #region 属性

        #endregion

        #region 委托事件

        #endregion

        #region 构造方法

        #endregion

        #region 私有方法

        #endregion

        #region 保护方法

        #endregion

        #region 公开方法
        /// <summary>
        /// 内存快照
        /// </summary>
        /// <param name="des"></param>
        public static void Snap(string des = null)
        {
            if (string.IsNullOrEmpty(des))
            {
                des = "app";
            }
            int avaiMem = Device.Instance.AvaiMem;
            if (lastAvaiMem < 1)
            {
                iTrace.Log("Loong", string.Format("{0}, AvaiMem:{1}M", des, avaiMem));
            }
            else
            {
                int dif = 0;
                string str = null;
                if (avaiMem < lastAvaiMem)
                {
                    dif = lastAvaiMem - avaiMem;
                    str = " ,decreased:";
                }
                else
                {
                    dif = avaiMem - lastAvaiMem;
                    str = " ,increased:";
                }
                iTrace.Log("Loong", string.Format("{0}, AvaiMem:{1}M ,{2}{3}", des, avaiMem, str, dif));
            }
        }
        #endregion
    }
}